File: src/main/java/net/hectus/neobb/turn/legacy_game/warp/LTDesertWarp.java
package net.hectus.neobb.turn.legacy_game.warp;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import net.hectus.neobb.turn.default_game.attributes.clazz.Clazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.HotClazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.NeutralClazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.RedstoneClazz;
import org.bukkit.World;
import org.bukkit.potion.PotionEffectType;

import java.util.List;

public class LTDesertWarp extends LWarpTurn {
    public LTDesertWarp(World world) { super(world, "legacy-desert"); }
    public LTDesertWarp(PlacedStructure data, World world, NeoPlayer player) { super(data, world, player, "legacy-desert"); }

    @Override
    public void apply() {
        super.apply();
        new Buff.Luck(Buff.BuffTarget.YOU, 5);
        new Buff.Effect(Buff.BuffTarget.OPPONENTS, PotionEffectType.SLOWNESS, 1);
    }

    @Override
    public int chance() {
        return 80;
    }

    @Override
    public List<Class<? extends Clazz>> allows() {
        return List.of(NeutralClazz.class, HotClazz.class, RedstoneClazz.class);
    }

    @Override
    public int cost() {
        return 4;
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/legacy_game/warp/LTDesertWarp.kt
package net.hectus.neobb.turn.legacy_game.warp

import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.Effect
import net.hectus.neobb.buff.Luck
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.structure.PlacedStructure
import net.hectus.neobb.turn.default_game.attribute.clazz.Clazz
import net.hectus.neobb.turn.default_game.attribute.clazz.HotClazz
import net.hectus.neobb.turn.default_game.attribute.clazz.NeutralClazz
import net.hectus.neobb.turn.default_game.attribute.clazz.RedstoneClazz
import org.bukkit.potion.PotionEffectType
import kotlin.reflect.KClass

class LTDesertWarp(data: PlacedStructure?, player: NeoPlayer?) : LWarpTurn(data, "desert", player) {
    override val cost: Int = 4
    override val chance: Double = 80.0
    override val allows: List<KClass<out Clazz>> = listOf(NeutralClazz::class, HotClazz::class, RedstoneClazz::class)
    override val temperature: Temperature = Temperature.HOT

    override fun apply() {
        Luck(5).apply(player!!)
        Effect(PotionEffectType.SLOWNESS, target = Buff.BuffTarget.OPPONENTS).apply(player)
    }
}